### PR TITLE
Add block_tags option

### PR DESCRIPTION
**Pull request by stevenbrookes**
**What this PR does**:
Adds the ability to set tags when blocks are saved to S3 storage.  This is useful for creating lifecycles in S3.
https://community.grafana.com/t/configuration-for-s3-intelligent-tiering/66028

**Which issue(s) this PR fixes**:

**Checklist**
- [ ] Tests updated
- [*] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/tempo/website/configuration/_index.md [hunk 1]
```diff
@@ -550,6 +550,13 @@ storage:
             # The maximum number of requests to execute when hedging. Requires hedge_requests_at to be set.
             [hedge_requests_up_to: <int>]
 
+              # Optional
+              # Example: "block_tags: {'key': 'value'}"
+              # A map key value strings for user tags to store on the S3 block objects, to aid setup of filters in S3 lifecycles.
+              # The tags are not set on index.json.gz files as these are rewritten often.
+              # See the S3 documentation for more detail: https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html
+              [block_tags: <map[string]string>]
+
```
Comment (knylander-grafana) [inline docs/tempo/website/configuration/_index.md:559]:
```suggestion
           # Optional
           # Example: "block_tags: {'key': 'value'}"
           # A map key value string for user tags to store on the S3 block objects, to aid the setup of filters in S3 lifecycles.
           # The tags are not set on index.json.gz files as these are rewritten often.
           # See the [S3 documentation on object tagging](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html) for more detail. 
              [block_tags: <map[string]string>]

```

**Comment by zalegrala** [comment]
This looks pretty good to me, thanks for the contribution.  I think it would be good for us to include an end-to-end test configuration update that we could verify against the minio instance that runs as part of the integration tests.  In that same vein, perhaps including a unit test similar to what we have for the GCS metadata would also be valuable here.

https://github.com/grafana/tempo/blob/main/tempodb/backend/gcs/gcs_test.go#L104

What do you think about adding those couple of test updates?

**Comment by stevenbrookes** [comment]
I agree a test would be ideal.  I had a quick look at gcs_test.go and I think it'll be a stretch for me to do that as not a native go developer yet.

**Comment by zalegrala** [comment]
Okay no sweat.  I'll find some time this week to write a test or two.  Do you mind if I push to this branch?

**Comment by zalegrala** [comment]
I've put up https://github.com/stevenbrookes/tempo/pull/1 to merge into this branch.

**Review by knylander-grafana — APPROVED** [review]
Documentation update looks good.

File: docs/tempo/website/configuration/_index.md [hunk 2]
```diff
@@ -550,6 +550,13 @@ storage:
...
             [hedge_requests_up_to: <int>]
 
+           # Optional
+           # Example: "block_tags: {'key': 'value'}"
+           # A map key value string for user tags to store on the S3 block objects. This string helps set up filters in S3 lifecycles.
+           # The tags are not set on index.json.gz files since these files are rewritten often.
+           # See the [S3 documentation on object tagging](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html) for more detail. 
+            [block_tags: <map[string]string>]
+
         # azure configuration. Will be used only if value of backend is "azure"
```
Comment (knylander-grafana) [inline docs/tempo/website/configuration/_index.md:560]:
```suggestion
        # Azure configuration. Used only if the value of backend is "azure".
```
```suggestion
        # azure configuration. Will be used only if value of backend is "azure"
```

**Pull request closed without merging**